Modern/wrapperPlugins/src/main/java/org/bonej/wrapperPlugins/AnalyseSkeletonWrapper.java

```diff
@@ -319,15 +321,18 @@ private void showAdditionalResults(final SkeletonResult results) {
 				((IntColumn) columns.get(1)).add((j + 1));
 				((DoubleColumn) columns.get(2)).add((edge.getLength()));
 				final Point point = edge.getV1().getPoints().get(0);
-				((IntColumn) columns.get(3)).add((point.x));
-				((IntColumn) columns.get(4)).add((point.y));
-				((IntColumn) columns.get(5)).add((point.z));
+				((DoubleColumn) columns.get(3)).add((point.x) * cal.pixelWidth);
+				((DoubleColumn) columns.get(4)).add((point.y) * cal.pixelHeight);
+				((DoubleColumn) columns.get(5)).add((point.z) * cal.pixelDepth);
 				final Point point2 = edge.getV2().getPoints().get(0);
-				((IntColumn) columns.get(6)).add((point2.x));
-				((IntColumn) columns.get(7)).add((point2.y));
-				((IntColumn) columns.get(8)).add((point2.z));
-				final double distance = MathArrays.distance(new double[] { point.x,
-					point.y, point.z }, new double[] { point2.x, point2.y, point2.z });
+				((DoubleColumn) columns.get(6)).add((point2.x) * cal.pixelWidth);
+				((DoubleColumn) columns.get(7)).add((point2.y) * cal.pixelHeight);
+				((DoubleColumn) columns.get(8)).add((point2.z) * cal.pixelDepth);
+				final double distance = MathArrays.distance(
+					new double[] { point.x * cal.pixelWidth,
+						point.y * cal.pixelHeight, point.z * cal.pixelDepth },
+					new double[] { point2.x * cal.pixelWidth, 
+						point2.y * cal.pixelHeight, point2.z * cal.pixelDepth });
```

Comment:
Could have a helper method
```
// Calibration is a private field
	private Vector3dc getCalibratedPoint(Vertex vertex) {
		final Point point = vertex.getPoints().get(0);
		final double px = point.x * calibration.pixelWidth;
		final double py = point.y * calibration.pixelHeight;
		final double pz = point.z * calibration.pixelDepth;
		return new Vector3d(px, py, pz);
	}
```
Then as a bonus you'd get distance just from `point1.distance(point2)`

Reply:
> Could have a helper method

`AnalyzeSkeleton3D_` has exactly what you suggest, but it's private (and inefficiently implemented with `Math.pow`). I made a PR to fix that, https://github.com/fiji/AnalyzeSkeleton/pull/16

I can see that this method should be both `static` and maybe in another class entirely, but refactoring would open a whole can of worms that none of us can prioritise right now.